docs/proposals/going-towards-to-crd.md

```diff
@@ -0,0 +1,104 @@
+
+## Alternative Solution
+
+Before merging the CRDs solution into the master, create a branch `release-0.1` with the last release of Service Catalog with Aggregated API server. 
+On the `release-0.1` branch we are still doing the **bug fixing** for this version of Service Catalog but features are not introduced. When some bug will be found then we can fix it and still in an easy way create a release with version `0.1.x` 
+
+In the master branch, we have Service Catalog with the CRD approach. New releases are created with the `0.2.x` version. On master branch, we are doing both the bug fixing and adding new features.
+
+Such strategy will help us to get new customers cause we will show that the CRD is our new direction on which we want to focus. 
+ 
+On the other hand, existing Service Catalog customers will see that the old solution is deprecated and the new features will be available only in the new approach. Thanks to that they will consider updating their system to the newest version as soon as possible cause the goal is to use the CRD solution as Kubernetes community recommends that.
+
+We can set a reasonable time for supporting bug fixing for the Service Catalog with Aggregated API, e.g. 1 year.
+
+With such solution, problems described in the previous section are mitigated.  
+     
+## Migration Support
+
+If we want to convince existing customers to use the new Service Catalog version then we need to create a migration guide/scripts. The migration needs to be as simple as possible.
+
+The migration will be simpler when we will hardcode for a given functionality in Aggregated API. 
+
+#### Details
+
+The migration logic can be placed in the Service Catalog Helm Chart. 
+When someone wants to upgrade existing Service Catalog to the newest one, then he can execute `helm upgrade {release-name} svc-cat/catalog`
+
+Raw scenario:
+- **pre-upgrade hook**
+  - replace the api-server deployment with the api-server that has a read-only mode
+  - backup the Service Catalog resources to the persistent volume
+- **upgrade**
+  - removes the api-server
+  - remove the etcd storage
+  - adjust secrets, RBAC etc.
+  - upgrade controller manager
+  - install webhook server
+- **post-upgrade**
+  - scale down the controller manager to 0  
+  - restore the Service Catalog resources - Spec and Status (status is important cause we do not want to trigger provisioning for already process items)
+  - scale up controller manager
+
+
+## Sum up
+
+We need to be sure that customer will exactly know in which direction we want to go. Supporting only bug fixing for api-server could simplify adding new features while we will still provide support for older clients. The migration process will also be easier when we will hardcode for a given functionality in Aggregated API. For new contributors, it will be much easier to get familiar with the code base where only one approach is used. 
```

Comment:
```suggestion
We need to be sure that customer will exactly know in which direction we want to go and why. Supporting only bug fixing for api-server will simplify adding new features and will give time to customers for migration. The migration process will be easier when we will stop developing features in the Aggregated API Server solution. For new contributors, it will be much easier to get familiar with the code base where only one approach is used. 
```